### DISCUSSION TITLE
UX annoyances

### DISCUSSION THREAD

**Discussion by aronaronaronaronaron** [godotengine/godot-proposals#6554] [Editor]
### Describe the project you are working on

i don't think it's neccessary to describe this project, because this issue is about the newest editor's simple changes

### Describe the problem or limitation you are having in your project

scripting, and using the editor has become a tiny bit harder thanks to the new 4.0 update

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

- move the transform options from layout, or give the ability to move the options, or, the simplest option, just put the options below all the other layout options

![image](https://user-images.githubusercontent.com/60071448/226739650-31910872-5c85-4a5a-bc66-4131206d9eae.png)

> this is probably the most used option, and this update has made it twice as hard to access it.

- don't delete code after auto-finished code

![image](https://user-images.githubusercontent.com/60071448/226738048-cf4c0fd1-c593-4a90-9d84-77af19b655bf.png)
>  when i press tab, i'm usually working on code. and working on a line of code doesn't just happen for the first time, so when everything i already wrote gets deleted cause i went back to change something little, it's obvious that could make someone mad.

- ctrl+d text duplication

![image](https://user-images.githubusercontent.com/60071448/226738819-aaf4821e-6188-40c9-b4b0-195dcb1b1ea8.png)
> two cursors are probably going to be useful down the line, but copy and pasting text with 1 key combo is more important, at least for me.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

basically just how godot 3 worked

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it will be used often

### Is there a reason why this should be core and not an add-on in the asset library?

i think these changes are probably universal enough to change

sorry btw for the earlier issue and this issue's wording, i wrote the last one ironically and i'm still annoyed here.

**Comment by KoBeWi** [comment]
> ctrl+d text duplication

The shortcut was changed to Ctrl+Shift+D, you can change it back in editor settings.

**Reply by aXu-AP** [reply]
What was the reasoning behind this? I find it strange to use `ctrl+D` to duplicate elsewhere in editor but for text it's `ctrl+shift+D`. Not that it's a big problem for me, I've changed it long ago in settings (along with `shift+del` for delete line), but newcomers might find these funky. Well, at least it's not something like `ctrl+J` (👀 at Photoshop).

Op seems to be aware of `ctrl+shift+D` shortcut as they mentioned specifically 1 key combo being better (and which I agree):
>two cursors are probably going to be useful down the line, but copy and pasting text with 1 key combo is more important, at least for me.

**Reply by AThousandShips** [reply]
Change is from godotengine/godot#48739

**Reply by aXu-AP** [reply]
Thanks! This is one of those things that differ depending of where the user comes from and what is their preferred writing style. I for example have never used a shortcut to select a whole word - I opt to double-click with a mouse if it is too long a word to select with `shift+arrows`. Duplicating lines is something I do very often as code usually requires similiar lines with small differences and sometimes just to make a quick backup of a block of code and then modify it.

However, some people seem to have it other way around and for them selecting words is a lifesaver but duplicating selection is less important (as seen in that pr).

**Reply by AThousandShips** [reply]
Important to remember that mouse use isn't always ergonomic, accessible, or efficient, and many prefer to be able to accomplish such functionality with keyboard only

**Reply by KoBeWi** [reply]
> Op seems to be aware of ctrl+shift+D shortcut as they mentioned specifically 1 key combo being better (and which I agree):

Well, the shortcut can be easily changed.

Also, before it was Ctrl+D, it was using Ctrl+B shortcut and was changed when bookmarks were added.

**Reply by aXu-AP** [reply]
Yes, grabbing the mouse just to select a word is definitely inferior to shortcut, if you don't happen to have mouse already in hand (or happen to use roller mouse). I might try to incorporate select word hotkey into my workflow - I never gave it a fair chance due to it taking my beloved duplicate line shortcut away 😆 

Edit: Managing shortcuts in a software which gets improved over time must be a nightmare. Never will be everybody satisfied - changes quickly raises some eyebrows but not responding to growing feature set also results in unintuitive shortcuts. Again, Photoshop seems to be one especially bad example of this with very little logic in a lot of shortcuts - the most courageous jump there in ages was making `ctrl+Z` to do regular undo (before it was a single step undo/redo).
